cmd/snap-seccomp/main.go

```diff
@@ -0,0 +1,522 @@
+	"NETLINK_ROUTE":          syscall.NETLINK_ROUTE,
+	"NETLINK_USERSOCK":       syscall.NETLINK_USERSOCK,
+	"NETLINK_FIREWALL":       syscall.NETLINK_FIREWALL,
+	"NETLINK_SOCK_DIAG":      C.NETLINK_SOCK_DIAG,
+	"NETLINK_NFLOG":          syscall.NETLINK_NFLOG,
+	"NETLINK_XFRM":           syscall.NETLINK_XFRM,
+	"NETLINK_SELINUX":        syscall.NETLINK_SELINUX,
+	"NETLINK_ISCSI":          syscall.NETLINK_ISCSI,
+	"NETLINK_AUDIT":          syscall.NETLINK_AUDIT,
+	"NETLINK_FIB_LOOKUP":     syscall.NETLINK_FIB_LOOKUP,
+	"NETLINK_CONNECTOR":      syscall.NETLINK_CONNECTOR,
+	"NETLINK_NETFILTER":      syscall.NETLINK_NETFILTER,
+	"NETLINK_IP6_FW":         syscall.NETLINK_IP6_FW,
+	"NETLINK_DNRTMSG":        syscall.NETLINK_DNRTMSG,
+	"NETLINK_KOBJECT_UEVENT": syscall.NETLINK_KOBJECT_UEVENT,
+	"NETLINK_GENERIC":        syscall.NETLINK_GENERIC,
+	"NETLINK_SCSITRANSPORT":  syscall.NETLINK_SCSITRANSPORT,
+	"NETLINK_ECRYPTFS":       syscall.NETLINK_ECRYPTFS,
+	"NETLINK_RDMA":           C.NETLINK_RDMA,
+	"NETLINK_CRYPTO":         C.NETLINK_CRYPTO,
+	"NETLINK_INET_DIAG":      C.NETLINK_INET_DIAG, // synonymous with NETLINK_SOCK_DIAG
+}
+
+const (
+	ScmpArchX86     = C.SCMP_ARCH_X86
+	ScmpArchX86_64  = C.SCMP_ARCH_X86_64
+	ScmpArchARM     = C.SCMP_ARCH_ARM
+	ScmpArchAARCH64 = C.SCMP_ARCH_AARCH64
+	ScmpArchPPC     = C.SCMP_ARCH_PPC
+	ScmpArchPPC64LE = C.SCMP_ARCH_PPC64LE
+	ScmpArchS390X   = C.SCMP_ARCH_S390X
+
+	SeccompRetAllow = C.SECCOMP_RET_ALLOW
+	SeccompRetKill  = C.SECCOMP_RET_KILL
+)
+
+// important for unit testing
+type SeccompData C.kernel_seccomp_data
+
+func (sc *SeccompData) SetNr(nr seccomp.ScmpSyscall) {
+	sc.nr = C.int(C.htot32(C.__u32(sc.arch), C.__u32(nr)))
+}
+func (sc *SeccompData) SetArch(arch uint32) {
+	sc.arch = C.htot32(C.__u32(arch), C.__u32(arch))
+}
+func (sc *SeccompData) SetArgs(args [6]uint64) {
+	for i := range args {
+		sc.args[i] = C.htot64(sc.arch, C.__u64(args[i]))
+	}
+}
+
+func readNumber(token string) (uint64, error) {
+	if value, ok := seccompResolver[token]; ok {
+		return value, nil
+	}
+
+	return strconv.ParseUint(token, 10, 64)
+}
+
+func parseLine(line string, secFilter *seccomp.ScmpFilter) error {
+	line = strings.TrimSpace(line)
+
+	// ignore comments and empty lines
+	if strings.HasPrefix(line, "#") || line == "" {
+		return nil
+	}
+
+	// regular line
+	tokens := strings.Fields(line)
+
+	// fish out syscall
+	secSyscall, err := seccomp.GetSyscallFromName(tokens[0])
+	if err != nil {
+		// FIXME: add structed error to libseccomp-golang
+		if err.Error() == "could not resolve name to syscall" {
+			return nil
+		}
+		return fmt.Errorf("cannot resolve name: %s", err)
+	}
+
+	var conds []seccomp.ScmpCondition
+	if err != nil {
+		return fmt.Errorf("cannot create new filter: %s", err)
+	}
+
+	for pos, arg := range tokens[1:] {
+		var cmpOp seccomp.ScmpCompareOp
+		var value uint64
+		var err error
+
+		if pos >= ScArgsMaxlength {
+			return fmt.Errorf("too many tokens (%d) in line %q", pos, line)
+		}
```

Comment:
We can check for this more efficiently up above, so if you implement that, feel free to remove this.